src/gtimelog/email.py

```diff
@@ -0,0 +1,85 @@
+import collections
+import email
+import logging
+import smtplib
+from contextlib import closing
+from email.mime import text
+from email.utils import formataddr, parseaddr
+from gettext import gettext as _
+
+from gtimelog import DEBUG, __version__
+
+
+MailProtocol = collections.namedtuple('MailProtocol', 'factory, startssl')
+MAIL_PROTOCOLS = {
+    'SMTP': MailProtocol(smtplib.SMTP, False),
+    'SMTPS': MailProtocol(smtplib.SMTP_SSL, False),
+    'SMTP (StartTLS)': MailProtocol(smtplib.SMTP, True),
+}
+
+log = logging.getLogger('email')
+
+
+class EmailError(Exception):
+    pass
+
+
+def isascii(s):
+    return all(0 <= ord(c) <= 127 for c in s)
+
+
+def address_header(name_and_address):
+    if isascii(name_and_address):
+        return name_and_address
+    name, addr = parseaddr(name_and_address)
+    name = str(email.header.Header(name, 'UTF-8'))
+    return formataddr((name, addr))
+
+
+def subject_header(header):
+    if isascii(header):
+        return header
+    return email.header.Header(header, 'UTF-8')
+
+
+def prepare_message(sender, recipient, subject, body):
```

Comment:
The tests for all the moved functions should also be moved into `test_email.py`.

It may be prudent to do the pytest switch before we start moving code (and tests) into new modules.